packages/itwinui-react/src/core/LabeledSelect/LabeledSelect.types-test.tsx

```diff
@@ -0,0 +1,68 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+/**
+ * This test file tests some LabeledSelect type related cases that have not been tested in other files
+ * (e.g. react-workshop, unit tests, etc.)
+ */
+
+import React, { useRef } from 'react';
+import LabeledSelect from './LabeledSelect.js';
+
+() => {
+  const ref = useRef(null);
+  return (
+    <>
+      <LabeledSelect
+        options={[
+          { value: 1, label: 'Option 1' },
+          { value: 2, label: 'Option 2' },
+          { value: 3, label: 'Option 3' },
+        ]}
+        onChange={(value) => {
+          const returnValue: number = value;
+          return returnValue;
+        }}
+        ref={ref}
+      />
+    </>
+  );
+};
+
+() => {
+  return (
+    <>
+      <LabeledSelect<number>
+        options={[
+          { value: 1, label: 'Option 1' },
+          { value: 2, label: 'Option 2' },
+          { value: 3, label: 'Option 3' },
+        ]}
+        onChange={(value) => {
+          const returnValue: number = value;
+          return returnValue;
+        }}
+      />
+    </>
+  );
+};
+
+() => {
+  return (
+    <>
+      <LabeledSelect<string>
+        label='Select Label'
+        options={[
+          { value: '1', label: 'Item #1' },
+          { value: '2', label: 'Item #2' },
+          { value: '3', label: 'Item #3' },
```

Comment:
I think you can also test that once a generic value is passed, the values are forced to be of that type.

E.g. you can provide the generic type of `string` but pass values of `number`. Then, you can add the comment with `@ts-expect-error` above the lines that cause the error. Example:

https://github.com/iTwin/iTwinUI/blob/0ccd7b7b1a8933becdd5c320756c13d773207d94/packages/itwinui-react/src/react-table/react-table.types-test.tsx#L87-L94